src/main/java/org/springframework/hateoas/server/core/UriTemplateFactory.java

```diff
@@ -37,7 +38,7 @@
 	 * @param mapping must not be {@literal null} or empty.
 	 * @return
 	 */
-	public static UriTemplate templateFor(String mapping) {
+	public static UriTemplate templateFor(@Nullable String mapping) {
```

Comment:
Because at least one use case depicts a nullable method being fed into it. So the input can receive a `null`, but it will catch it through the assertion and alert the user.

These situations are annoying, but not all can be eliminated.

Reply:
But then we still shouldn't have a `@Nullable` annotation on the parameter, should we?
After all the calling code is in error in passing `null` and should, therefore, be marked by code inspections.